OJS.Workers/LocalWorkerServiceBase.cs

```diff
@@ -84,16 +85,36 @@ protected virtual void BeforeAbortingThreads()
 
         private void SpawnSubmissionProcessorsAndThreads()
         {
-            var submissionsForProcessing = new ConcurrentQueue<TSubmission>();
+            var localWorkerSubmissionsForProcessing = new ConcurrentQueue<TSubmission>();
+            var legacyWorkersSubmissionsForProcessing = new ConcurrentQueue<TSubmission>();
+            var alphaWorkersSubmissionsForProcessing = new ConcurrentQueue<TSubmission>();
             var sharedLockObject = new object();
 
             var workerThreads = new List<(SubmissionProcessor<TSubmission> submissionProcessor, Thread thread)>();
 
             var formatterServiceFactory = new FormatterServiceFactory();
             var remoteSubmissionsFilteringService = new RemoteSubmissionsFilteringService();
             var localSubmissionsFilteringService = new LocalSubmissionsFilteringService();
-            workerThreads.AddRange(this.GetLocalWorkers(Settings.ThreadsCount, submissionsForProcessing, sharedLockObject, localSubmissionsFilteringService));
-            workerThreads.AddRange(this.GetRemoteWorkers(Settings.RemoteWorkerEndpoints, submissionsForProcessing, sharedLockObject, formatterServiceFactory, remoteSubmissionsFilteringService));
+            workerThreads.AddRange(this.GetLocalWorkers(
+                Settings.ThreadsCount,
+                localWorkerSubmissionsForProcessing,
+                sharedLockObject,
+                localSubmissionsFilteringService,
+                new List<WorkerType>{WorkerType.Local}));
+            
+            workerThreads.AddRange(this.GetRemoteWorkers(
+                Settings.RemoteWorkerEndpoints,
+                legacyWorkersSubmissionsForProcessing,
+                sharedLockObject, formatterServiceFactory,
+                remoteSubmissionsFilteringService,
+                new List<WorkerType>{WorkerType.Default, WorkerType.Legacy}));
```

Comment:
The `WorkerType.Default` should be passed to different threads, depending what is configured in App.config to be the default worker type. This way we are stuck with Legacy, until we make code change.
Do you need the `DefaultWorkerType` in App.config at all with this approach?

Reply:
Fixed.